### PR TITLE
ISPN-6593 Adjust JNI LevelDB for Windows

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-6593

Windows adjustments for LevelDB and some reformatting. Highlights:
* Most of the fixes should go to LevelDB itself, that's why they look a bit ugly in our code
* I tried to limit the scope of changes (that's why I often check if an instance is a JNI and do something slightly differently).

To both `master` and `8.2.x` please.
TIP: Review files with "ignore whitespaces" on: https://github.com/infinispan/infinispan/pull/4309/files?w=1

### CODE REVIEW

File: persistence/leveldb/src/main/java/org/infinispan/persistence/leveldb/LevelDBStore.java [hunk 1]
```diff
@@ -30,569 +19,638 @@
...
             }
-
-            eacs.waitUntilAllCompleted();
-            if (eacs.isExceptionThrown()) {
-               throw new PersistenceException("Execution exception!", eacs.getFirstException());
+        }
+
+    }
+
+    private void deleteFilesFromDirectory(File directory) {
```
Comment (tristantarrant) [inline persistence/leveldb/src/main/java/org/infinispan/persistence/leveldb/LevelDBStore.java]: TestingUtil.recursiveFileRemove does the same (aside from the GC), so maybe it's a good idea to promote it to commons main and reuse it here.

Comment (slaskawi) [inline persistence/leveldb/src/main/java/org/infinispan/persistence/leveldb/LevelDBStore.java]: Good idea, let me refactor it.

**Comment by tristantarrant** [comment]
Pushed to master and 8.2.x

**Comment by anistor** [comment]
The refactoring has lead to several unused imports of Util.recursiveFileRemove. I fixed that and pushed.

**Comment by anistor** [comment]
InfinispanRhqTest had an import issue on 8.2.x. Fixed and pushed.

**Comment by slaskawi** [comment]
@anistor Thanks a lot Adrian! I will open a PR against each branch next time.

**Pull request closed without merging**